HULKs/hulk
# Look at panel for Twix from Maxi's work

## Introduced Changes

Work is taken from https://github.com/schmidma/nao/tree/calibration-mode with reduced scope for this PR.

* Add a Look-At panel for Twix with minimal functionality inject motion command `LookAt`
* Can select which camera to look at or `None` for automatic selection
* Shortcut to look at penalty box from the centre (current calibration workflow), use a subscription for field dimensions.

![image](https://github.com/HULKs/hulk/assets/6742239/f86b26b1-f2f5-4526-85cd-bce22ddc4aef)


Fixes #115 

## ToDo / Known Issues

- [ ] Review if the UI layout is acceptable compared to the original proposal in #115 (https://user-images.githubusercontent.com/6742239/234351961-1425a3ec-09ab-46d5-9d49-38333c4b5e8b.png)
- [x] Check the current motion command/ state to verify the NAO is standing up and stable before sending commands.
## Ideas for Next Iterations (Not This PR)
* Add a clickable rectangle (top-view) to quickly dispatch a look-at, could be coupled with map overlay.
* Possible UI/ UX improvements
* Mode to trigger a sequence of motions (i.e. look at a grid of target points) -> capture data for extrinsic calibration and other vision tunings.

## How to Test

* Load the panel with Twix, **Make sure the NAO stands before clicking "enable motion override"**.
    * Motion override will send a standing lookAt command, if the NAO is not standing up already, it can cause a disastrous motion.
* Select manual or shortcut for the penalty box and "send command" button.
